engine/source/output/h3d/h3d_results/genh3d.F

```diff
@@ -923,57 +925,49 @@ SUBROUTINE GENH3D(TIMERS,X        ,D        ,V           ,A           ,BUFEL
-              ENDIF
 
-              !FVDATA data structure in a 1D array
-              NPOLH = MAX(1, NPOLH)
-              ALLOCATE(FVDATA_1D_ARRAY(12*NPOLH))
-
-              IF(NPOLH > 0)THEN
-                FVDATA_1D_ARRAY(00*NPOLH+1:01*NPOLH) = FVDATA(J)%CENTROID_POLH(1,1:NPOLH)
-                FVDATA_1D_ARRAY(01*NPOLH+1:02*NPOLH) = FVDATA(J)%CENTROID_POLH(2,1:NPOLH)
-                FVDATA_1D_ARRAY(02*NPOLH+1:03*NPOLH) = FVDATA(J)%CENTROID_POLH(3,1:NPOLH)
-                FVDATA_1D_ARRAY(03*NPOLH+1:04*NPOLH) = FVDATA(J)%QPOLH(1,1:NPOLH)
-                FVDATA_1D_ARRAY(04*NPOLH+1:05*NPOLH) = FVDATA(J)%QPOLH(2,1:NPOLH)
-                FVDATA_1D_ARRAY(05*NPOLH+1:06*NPOLH) = FVDATA(J)%QPOLH(3,1:NPOLH)
-                FVDATA_1D_ARRAY(06*NPOLH+1:07*NPOLH) = FVDATA(J)%PPOLH(1:NPOLH)
-                FVDATA_1D_ARRAY(07*NPOLH+1:08*NPOLH) = FVDATA(J)%SSPPOLH(1:NPOLH)
-                FVDATA_1D_ARRAY(08*NPOLH+1:09*NPOLH) = FVDATA(J)%DTPOLH(1:NPOLH)
-                FVDATA_1D_ARRAY(09*NPOLH+1:10*NPOLH) = FVDATA(J)%MPOLH(1:NPOLH)
-                FVDATA_1D_ARRAY(10*NPOLH+1:11*NPOLH) = FVDATA(J)%RPOLH(1:NPOLH)
-                FVDATA_1D_ARRAY(11*NPOLH+1:12*NPOLH) = FVDATA(J)%TPOLH(1:NPOLH)
-              ENDIF
-
-              ! spmd gathering of 1d array
-              NPOLHG = MAX(NPOLHG,1)
-              ALLOCATE(FVDATA_1D_ARRAY_P(12*NPOLHG))
-              FVDATA_1D_ARRAY_P(:) = 0
-              CALL SPMD_H3D_GATHER_R(FVDATA_1D_ARRAY, 12*NPOLH,FVDATA_1D_ARRAY_P, 12*NPOLHG)
-
-              ! on relevant domain (ispmd==0) retrieve FVBAG data structure from 1d array
-              IF(ISPMD==0)THEN
+              ! get NPOLHG,NPOLH_ANIM_G, AIRBAGS_TOTAL_FVM_IN_H3D
+              if ( (FVSPMD(J)%PMAIN-1 == 0) .and. (ispmd == 0)) then ! no communication Internal mesh is already on Domain 0
+
+                 NPOLHG                     = FVDATA(J)%NPOLH
+                 NPOLH_ANIM_G               = FVDATA(J)%NPOLH_ANIM
+                 AIRBAGS_TOTAL_FVM_IN_H3D_G = AIRBAGS_TOTAL_FVM_IN_H3D
+
+              else
+                 if (FVSPMD(J)%PMAIN-1 == ispmd)then ! the airbag is on another domain the ispmd=0
+
+                   FVM_GLOBALS(1) = FVDATA(J)%NPOLH
+                   FVM_GLOBALS(2) = FVDATA(J)%NPOLH_ANIM
+                   FVM_GLOBALS(3) = AIRBAGS_TOTAL_FVM_IN_H3D
+
+                   call spmd_send(FVM_GLOBALS,3,IT_SPMD(1),25001)
+                 endif
+ 
+                 if (ispmd == 0)then
+                    call spmd_recv(FVM_GLOBALS,3,IT_SPMD(FVSPMD(J)%PMAIN),25001)
+                    NPOLHG                     = FVM_GLOBALS(1)
+                    NPOLH_ANIM_G               = FVM_GLOBALS(2)
+                    AIRBAGS_TOTAL_FVM_IN_H3D_G = FVM_GLOBALS(3)
+                 else
+                    NPOLHG = 0
+                    NPOLH_ANIM_G = 0
+                    AIRBAGS_TOTAL_FVM_IN_H3D_G = 0
+                 endif
+              endif
+
+              !Debug
+              !if (ispmd == 0) then
```

Comment:
some remaining commented print
could you remove ?

Reply:
Done :)